src/txn/txn_recover.c

```diff
@@ -633,6 +622,51 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
         WT_ERR(ret);
     }
 
+    /*
+     * We should check whether the history store file exists in the metadata or not. or not. If it
+     * does not, then we should not apply rollback to stable to each table. This might happen if
+     * we're upgrading from an older version. If it does exist in the metadata we should check that
+     * it exists on disk to confirm that it wasn't deleted between runs.
+     *
+     * This needs to happen after we recovery the metadata from the logs, otherwise we can end up in
+     * the situation where the metadata does actually contain the history store but we didn't yet
+     * recover the metadata.
```

Comment:
This comment needs something more, I'm not following exactly what it is trying to say. If you take out the middle it doesn't say anything: "This needs to happen after we recover the metadata because ... but we didn't yet recover the metadata." It isn't clear what `the situation` is. <thinking a bit>
Do you mean the situation that the metadata should contain the history store data but doesn't actually contain that entry on disk before recovery?